service/history/timerGate.go

```diff
@@ -104,8 +104,11 @@ func NewLocalTimerGate() LocalTimerGate {
 		for {
 			select {
 			case <-timer.timer.C:
+				select {
 				// re-transmit on gateC
-				timer.fireChan <- struct{}{}
+				case timer.fireChan <- struct{}{}:
+				default:
```

Comment:
Still new to go. Will this prevent a send to the `fireChan` channel if there is already something in the channel?

Reply:
Looks like you already answered this.

Reply:
yes, if the channel is full, then it means there is already a signal inside.

this class aims to send out signal after given time, (like a timer). if the receiving side of the channel is not polling a message in the channel, adding more does not make anything better.